sklearn/linear_model/coordinate_descent.py

```diff
@@ -729,6 +737,9 @@ def fit(self, X, y, check_input=True):
         self.coef_, self.dual_gap_ = map(np.squeeze, [coef_, dual_gaps_])
         self._set_intercept(X_offset, y_offset, X_scale)
 
+        # workaround since _set_intercept will cast self.coef_ into float64
+        self.coef_ = np.asarray(self.coef_, dtype=X.dtype)
```

Comment:
In my opinion, it would be confusing if I fit `np.float32` training data, and get prediction result with`np.float64`.

So I think we need to make `coef_` match the type.